iml-gui/crate/src/lib.rs

```diff
@@ -271,16 +276,29 @@ pub fn update(msg: Msg, model: &mut Model, orders: &mut impl Orders<Msg>) {
         Msg::Records(records) => {
             model.records = records;
 
+            let old = model.activity_health.clone();
             model.activity_health =
                 update_activity_health(&model.records.active_alert);
+            orders.proxy(Msg::Notification).send_msg(notification::generate(
+                &None,
+                &old,
+                &model.activity_health,
+            ));
         }
         Msg::RecordChange(record_change) => match record_change {
             warp_drive::RecordChange::Update(record) => match record {
```

Comment:
I don't think so. We need an updated activity health - when the current alert is accounted, i. e. after `model.records.active_alert.insert(x.id, x);`

Reply:
Ah, right. 

A pattern I've been using is to pass id's around instead of entire objects since `u32` implements `Copy`.

Then the `notification::update` could take a reference to `Cache` as the first param and the id can be resolved to a record inside of `notification::update`.

That said, having clone here is hopefully not too expensive.

Reply:
Actually, it looks like you only use the `message` field of `Alert`. Any reason you can't pass / clone that instead? It would be cheaper than cloning the entire `Alert`

Reply:
Yes, probably could be just String.  I used (Alert, ActivityHeath, ActivityHeath) as a maximum abstraction.

Reply:
I'd like to be able to pass in whole objects to sub-components, but haven't found a nice way to do it without cloning / copying

Reply:
Let it be a string for now.

Reply:
I'm going to explore using [immutable data structures](https://docs.rs/im/14.1.0) here, which may be an elegant way to solve this issue.